h2/src/main/org/h2/index/Index.java

```diff
@@ -527,6 +528,19 @@ public DbException getDuplicateKeyException(String key) {
         return e;
     }
 
+    private StringBuilder getIndexOrConstraintSQL(StringBuilder builder) {
+        ArrayList<Constraint> constraints = table.getConstraints();
+        if (constraints != null) {
+            for (Constraint constraint : table.getConstraints()) {
+                if (constraint.usesIndex(this) && constraint.getConstraintType() == Constraint.Type.UNIQUE) {
```

Comment:
`Type.PRIMARY_KEY` must be considered as well.

Reply:
Isn't primary key handled by getDuplicatePrimaryKeyMessage(), not  getDuplicateKeyException() ?

Reply:
Primary key constraints are backed by delegate or secondary index. `getDuplicatePrimaryKeyMessage()` is only used by a delegate index, secondary index uses `getDuplicateKeyException()`.

Reply:
```SQL
CREATE TABLE TEST(ID UUID CONSTRAINT TEST_PK PRIMARY KEY);
INSERT INTO TEST VALUES '65b05eac-c855-4413-963c-701dc35a744f', '65b05eac-c855-4413-963c-701dc35a744f';
```
```
Unique index or primary key violation: "PUBLIC.PRIMARY_KEY_2 ON PUBLIC.TEST(ID) VALUES ( /* 1 */ UUID '65b05eac-c855-4413-963c-701dc35a744f' )"
at org.h2.message.DbException.getJdbcSQLException(DbException.java:520)
    at org.h2.message.DbException.getJdbcSQLException(DbException.java:489)
    at org.h2.message.DbException.get(DbException.java:223)
    at org.h2.message.DbException.get(DbException.java:199)
    at org.h2.index.Index.getDuplicateKeyException(Index.java:525)
    at org.h2.mvstore.db.MVSecondaryIndex.checkUnique(MVSecondaryIndex.java:223)
```

Reply:
You right, I forgot that, thanks.